### DISCUSSION TITLE
How can I change the output binary file-name for my port?

### DISCUSSION THREAD

**Discussion by jdx-john** [microsoft/vcpkg#20099] [Q&A]
My port is now building but the DLL is named differently to how I would like. Specifically, I would like to include the architecture in the filename - but I cannot see how/if I can override this from my portfile?

The normal CMake way would be to use `set_target_properties` I think, but this gives the error "set_target_properties command is not scriptable".

I wondered if it might be one of these: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_common_definitions.md

But this made no difference when I tested:
```
#set(VCPKG_TARGET_SHARED_LIBRARY_PREFIX ${VCPKG_TARGET_ARCHITECTURE})
vcpkg_configure_cmake(
  SOURCE_PATH ${SOURCE_PATH}
  PREFER_NINJA
)
vcpkg_install_cmake()

```

My ideal would be to get a DLL/LIB named `<port-name><architecture>.lib/dll`; it seems like there is probably something in vcpkg to do this but I cannot find it.

**Comment by autoantwort** [comment]
You can patch the build system or do a `file(RENAME ...)` after `vcpkg_install_cmake()`.

**Reply by jdx-john** [reply]
Ah, so not something "out of the box". OK thanks, I didn't want to start hacking if there was a solution provided.
